### PR TITLE
Add hello example

### PR DESCRIPTION
**Pull request by CamilleEviden**

### CODE REVIEW

File: examples/hello/client.c [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	ret = PtlEQAlloc(nih, 10, &eqh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlEQAlloc failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto ni_fini;
+	}
+
+	/* describe the memory used to send the message */
+	md = (ptl_md_t){ .start = argv[2],
+			.length = strlen(argv[2])+1,
```
Comment (boyerq) [inline examples/hello/client.c:88]: I think this would not pass clang-format?

File: examples/hello/client.c [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+#include "portals4.h"
+#include "portals4_bxiext.h"
+
+#include "hello.h"
+
+/*
+ * this file represent a client sendind a message to a server
+ *  
+ * Usage: ./client [nid] [message]
```
Comment (boyerq) [inline examples/hello/client.c:32]:
```suggestion
 * Usage: ./client <nid> <message>
```

The difference between `[argument]` and `<argument>` is that the first one is optional, whereas the second one is mandatory

File: examples/hello/server.c [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	ptl_handle_eq_t eqh;
+	ptl_process_t id;
+	ptl_event_t ev;
+	ptl_handle_le_t leh;
+	long num_messages;
+	ptl_le_t le;
+	char *p;
+	char msg[PTL_EV_STR_SIZE];
+	char data[50];
+	int wait_for_ever = 0;
```
Comment (boyerq) [inline examples/hello/server.c:54]: Use a `bool` it's clearer

File: examples/hello/server.c [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+	/* expose receive buffer to the network */
+	ret = PtlLEAppend(nih, pti, &le, PTL_PRIORITY_LIST, NULL, &leh);
+	if (ret != PTL_OK) {
+		fprintf(stderr, "PtlLEAppend failed : %s \n", PtlToStr(ret, PTL_STR_ERROR));
+		res = 1;
+		goto pt_free;
+	}
+
+	ret = PtlEQWait(eqh, &ev);
```
Comment (pichong) [inline examples/hello/server.c:122]: could add a comment here to explain we are waiting for LINK event

File: examples/hello/server.c [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+		}
+		data[sizeof(data) - 1] = '\0';
+		PtlEvToStr(0, &ev, msg);
+		printf("Event : %s", msg);
+		if (ev.type == PTL_EVENT_PUT) {
+			if (ev.rlength + 1 >= sizeof(data)) {
+				fprintf(stderr,
+					"The received message is too long to be entirely readed\n");
+			} else {
+				((char *)ev.start)[ev.rlength + 1] = 0;
```
Comment (pichong) [inline examples/hello/server.c:149]: I don't think this is needed.
The \0 final string character has been transmitted as part of the message.

File: examples/put_to_self.c
Comment (pichong) [inline examples/put_to_self.c:1]: these changes should be made in another commit

**Review by pichong — COMMENTED** [review]
looks very good !

**Review by pichong — APPROVED** [review]
